### ISSUE TITLE
1.8.X fails to build on Fedora Rawhide for 32bit due to dcmtk

### ISSUE THREAD

**Issue by hobbes1069** [OpenImageIO/oiio#1841]
### Problem
During compilation on 32bit systems the following error occurs with the dicom library, dcmtk:
```
/builddir/build/BUILD/oiio-Release-1.8.7/src/dicom.imageio/dicominput.cpp: In member function 'void OpenImageIO_v1_8::DICOMInput::read_metadata()':
/builddir/build/BUILD/oiio-Release-1.8.7/src/dicom.imageio/dicominput.cpp:301:56: error: cannot bind non-const lvalue reference of type 'Sint32& {aka long int&}' to an rvalue of type 'Sint32 {aka long int}'
                 if (dataset->findAndGetSint32 (tag, val).good())
                                                        ^
In file included from /usr/include/dcmtk/dcmdata/dctk.h:50:0,
                 from /builddir/build/BUILD/oiio-Release-1.8.7/src/dicom.imageio/dicominput.cpp:40:
/usr/include/dcmtk/dcmdata/dcitem.h:814:17: note:   initializing argument 2 of 'OFCondition DcmItem::findAndGetSint32(const DcmTagKey&, Sint32&, long unsigned int, bool)'
     OFCondition findAndGetSint32(const DcmTagKey &tagKey,
                 ^~~~~~~~~~~~~~~~
/builddir/build/BUILD/oiio-Release-1.8.7/src/dicom.imageio/dicominput.cpp:305:56: error: cannot bind non-const lvalue reference of type 'Uint32& {aka long unsigned int&}' to an rvalue of type 'Uint32 {aka long unsigned int}'
                 if (dataset->findAndGetUint32 (tag, val).good())
                                                        ^
In file included from /usr/include/dcmtk/dcmdata/dctk.h:50:0,
                 from /builddir/build/BUILD/oiio-Release-1.8.7/src/dicom.imageio/dicominput.cpp:40:
/usr/include/dcmtk/dcmdata/dcitem.h:786:17: note:   initializing argument 2 of 'OFCondition DcmItem::findAndGetUint32(const DcmTagKey&, Uint32&, long unsigned int, bool)'
     OFCondition findAndGetUint32(const DcmTagKey &tagKey,
                 ^~~~~~~~~~~~~~~~
{standard input}: Assembler messages:
{standard input}: Error: .size expression for dicominput.cpp does not evaluate to a constant
```

Full build logs:
i686:
https://kojipkgs.fedoraproject.org//work/tasks/5776/23995776/build.log

armv7hl:
https://kojipkgs.fedoraproject.org//work/tasks/5778/23995778/build.log

### Versions

* OIIO branch/version:  1.8.7
* OS: Fedora Linux (Rawhide) 
* C++ compiler: 7.2.1

**Comment by lgritz** [comment]
Do you happen to know know which version of Dcmtk you have installed?

If the dctk.h header is the source of the error message, how did Dcmtk itself get compiled cleanly on that system?

**Comment by hobbes1069** [comment]
Version 3.6.2. I think there was a bad glibc package for a while but not sure if it's related. Let me try again...

**Comment by hobbes1069** [comment]
Nope, tried a fresh rebuild of dcmtk and OIIO and got the same error...

**Comment by lgritz** [comment]
@hobbes1069, I think I see what's going on now. Somehow Sint32 must not be the same as 'int' and Uint32 must not be the same as 'unsigned int.' I have a proposed fix in #1843. I know that's a patch against master, but I'll backport if it works. Can you give a try and see if it gives you a clean build on those platforms?
